docs/diagrams/MatchAutoActivityDiagram.puml

```diff
@@ -0,0 +1,19 @@
+@startuml
+skinparam defaultTextAlignment center
+skinparam DefaultFontSize 13
+
+start
+:User executes `match auto`;
+:Generate all possible matches;
+:Sort candidate matches
+by match score;
+repeat
+    :Examine next best candidate;
+    if () then ([Buyer and Property unmatched])
+        :Confirm match;
+    else ([Buyer or Property taken])
+        :Discard candidate;
+    endif
+repeat while () is ([candidates remaining])
+stop
```

Comment:
How about deciding the display order of the matches? In the case of the program, the sorting comparisons is slightly different from the sorting of the candidate matches.